scripts/linters/pylint_extensions.py

```diff
@@ -1893,6 +1893,204 @@ def visit_compare(self, node):
                 self.add_message('inequality-with-none', node=node)
 
 
+class DisallowedFunctionCallsChecker(checkers.BaseChecker):
+    """Custom pylint checker for language specific general purpose
+    regex checks of functions calls to be removed or replaced.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+    name = 'disallowed-function-calls'
+    priority = -1
+    msgs = {
+        'C0031': (
+            'Please do not use %s.',
+            'remove-disallowed-function-calls',
+            'Please remove disallowed function calls.',
+        ),
+        'C0032': (
+            'Please replace %s with %s.',
+            'replace-disallowed-function-calls',
+            'Please replace disallowed function' +
+            'calls with allowed alternative.',
+        ),
+    }
+    python_utils_file = b'python_utils.py'
+    python_utils_test_file = b'python_utils_test.py'
+
+    # pylint: disable=disallowed-function-calls,single-line-pragma
+    options = ((
+        'disallowed-functions',
+        {
+            'default': (
+                (
+                    re.compile(r'datetime.datetime.now\(\)'),
```

Comment:
why do we need regex here?

Reply:
I'm following `general_purpose_linter`'s implementation, but I try to do this without regex as well.